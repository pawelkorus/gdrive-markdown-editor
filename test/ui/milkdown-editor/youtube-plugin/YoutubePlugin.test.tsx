import { test, expect } from 'vitest'
import React from 'react'
import { render, act } from '@testing-library/react'
import { TestMilkdownEditor } from '@test/test-utils'

const SAMPLE_CONTENT = '::youtube{src="sampleyoutube"}'

test('should render youtube embed iframe', async () => {
  await act(async () => {
    render(<TestMilkdownEditor content={SAMPLE_CONTENT} />)

    await new Promise(r => setTimeout(r, 4000))
  })

  expect(document.body).toMatchSnapshot()
})
