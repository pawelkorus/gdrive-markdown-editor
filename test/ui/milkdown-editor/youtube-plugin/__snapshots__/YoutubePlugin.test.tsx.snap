// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render youtube embed iframe 1`] = `
<body>
  <script
    defer=""
    src="https://accounts.google.com/gsi/client"
  />
  <script
    defer=""
    src="https://apis.google.com/js/api.js"
  />
  <div>
    <div
      data-milkdown-root="true"
    >
      <div
        class="milkdown"
      >
        <div
          class="ProseMirror editor"
          contenteditable="true"
          role="textbox"
          translate="no"
        >
          <div
            data-node-view-root="true"
          >
            <p
              class="text-center ratio ratio-16x9"
              data-type="youtube"
            >
              <iframe
                allowfullscreen=""
                src="https://www.youtube.com/embed/sampleyoutube"
              />
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
